### DISCUSSION TITLE
[.clinkpromt] Can :rightfilter be used to place something on a "multiline" :filter line?

### DISCUSSION THREAD

**Discussion by Jerakin** [chrisant996/clink#684] [Q&A] [answered]
In [Pure](https://github.com/chrisant996/clink/blob/master/clink/app/themes/pure.clinkprompt)  `:filter` does `line1.."\n"..line2`. Can I use `:rightfilter` to place something on the same line as `line1` but on the right side? If so, how?

**Comment by chrisant996** [comment, marked as answer]
Look at how Antares, bureau, Headline, and jonathan add right-aligned text on the first line.

Inside `:filter()` they use `os.getwidth()` to find the width of the terminal, `console.cellcount()` to find the displayed width of text (counting the characters in a string is not the same as the displayed width), and `string.rep()` to make a string of spaces or line characters to fill the middle area and right-justify some text on the first line.

**Reply by Jerakin** [reply]
Thanks. That's what I thought needed to be done, but wanted to double check to be 100% sure.

Here's the outcome. https://gist.github.com/Jerakin/2bce5afb88e40b406780dc6019aaaaaf
